### ISSUE TITLE
fmeasure is not evaluated correctly

### ISSUE THREAD

**Issue by DepthFirstSearch** [fchollet/keras#4564]
Hello,

the metric `fmeasure` (or one of its aliases) is always the same value as the accuracy:

```
from keras.metrics import fmeasure
model = ...
model.compile(loss='categorical_crossentropy', optimizer='sgd', metrics=['accuracy', fmeasure])
hist = model.fit_generator(verbose=2, ...)
```
As you can see, `fmeasure` is always the same as `acc` as well as `val_fmeasure` and `val_acc`:
> Epoch 60/1000
> 6s - loss: 0.6053 - acc: 0.8737 - fmeasure: 0.8737 - val_loss: 0.4240 - val_acc: 0.8151 - val_fmeasure: 0.8151
> Epoch 61/1000
> 6s - loss: 0.5862 - acc: 0.8830 - fmeasure: 0.8830 - val_loss: 0.4657 - val_acc: 0.8082 - val_fmeasure: 0.8082
> Epoch 62/1000
> 6s - loss: 0.5992 - acc: 0.8779 - fmeasure: 0.8779 - val_loss: 0.4437 - val_acc: 0.8219 - val_fmeasure: 0.8219
> Epoch 63/1000
> 6s - loss: 0.6070 - acc: 0.8712 - fmeasure: 0.8712 - val_loss: 0.6948 - val_acc: 0.7466 - val_fmeasure: 0.7466

The same happens when I compile like:
```
model.compile(loss='categorical_crossentropy', optimizer='sgd', metrics=['accuracy', 'fscore'])
```

**Comment by carlthome** [comment]
Hmm, what's your data? How many labels do you have per input sample?

**Comment by DepthFirstSearch** [comment]
I tried the same on another machine of mine. There it's working correctly. So, the problem must be my installation. Sorry for the inconvenience!
